stdlib/domain.mli

```diff
@@ -27,7 +27,7 @@
 
     @since 5.0 *)
 
-type !'a t
+type !'a t : value mod portable contended with 'a
```

Comment:
Is this safe? It looks like it would let you call `join` from different capsules, exposing the `'a` into multiple capsules simultaneously even though it might be mutable.

Reply:
If `'a` is mutable then it shouldn't cross contention and then `'a t` also shouldn't cross contention and calling
```ocaml
val join : 'a t -> 'a @@ portable
```
from multiple capsules should be prevented.

Reply:
Sorry, I just missed the `with 'a` part. This is clearly fine.